gammapy/scripts/cta_sensitivity.py

```diff
@@ -110,47 +111,29 @@ def get_excess(self, bkg_counts):
 
         Notes
         -----
-        For the moment, search by dichotomy.
-
-        TODO: Cf. np.vectorize for an time optimisation of this function.
+        Find the number of needed gamma excess events using newtons method.
+        Defines a function `significance_on_off(x, off, alpha) - self.sigma`
+        and uses scipy.optimize.newton to find the `x` for which this function
+        is zero.
         """
-        excess = np.zeros(len(bkg_counts))
-        for icount in range(len(bkg_counts)):
 
-            if bkg_counts[icount]<1:
-                excess[icount]=self.gamma_min
-                continue
+        def target_function(on, off, alpha):
+            return significance_on_off(on, off, alpha, method='lima') - self.sigma
 
-            # Coarse search
-            start, stop = -1., 6.
-            coarse_excess = np.logspace(start=start, stop=stop, num=1000)
-            coarse_on = coarse_excess + bkg_counts[icount]
-            coarse_off = np.zeros(len(coarse_on)) + bkg_counts[icount] / self.alpha
-            coarse_sigma = significance_on_off(n_on=coarse_on, n_off=coarse_off, alpha=self.alpha, method='lima')
-            idx = np.abs(coarse_sigma - self.sigma).argmin()
-
-            start = coarse_excess[max(idx - 1, 0)]
-            stop = coarse_excess[min(idx + 1, len(coarse_sigma) - 1)]
-            if start == stop:
-                log.warning('LOGICAL ERROR> Impossible to find a number of gamma!')
-                excess[icount] = -1
+        excess = np.zeros_like(bkg_counts)
+        for energy_bin, bg_count in enumerate(bkg_counts):
```

Comment:
Suggest we change "bg" -> "bkg" for uniformity in naming here.
(and if needed in follow-up PRs in other parts of the Gammapy code as well, but here please keep changes to this one file / class)

Reply:
done